entgql/template.go

```diff
@@ -443,12 +443,19 @@ func orderFields(n *gen.Type) ([]*OrderTerm, error) {
 			})
 		}
 	}
+	edgeNamesWithThroughTablesSet := make(map[string]interface{})
 	for _, e := range n.Edges {
 		name := strings.ToUpper(e.Name)
 		switch ant, err := annotation(e.Annotations); {
 		case err != nil:
 			return nil, err
-		case ant.Skip.Is(SkipOrderField), ant.OrderField == "":
+		case ant.Skip.Is(SkipOrderField):
+			// skip edges marked with the skip annotation
+		case ant.OrderField == "":
+			// skip edges without the order field annotation
+		case strings.HasSuffix(ant.OrderField, "_COUNT") &&
+			edgeNamesWithThroughTablesSet[ant.OrderField[:len(ant.OrderField)-6]] != nil:
```

Comment:
Not sure I understand the logic here, but I think the current bug can be fixed on the template side.

Reply:
It was looking like an order field was being created for both the edge and the join table, but it's only needed for one.

I think if it's done in the templates we'd have to update it in more than one spot, but perhaps that'd be the more correct way to go about it.